### PR TITLE
TXAD-1: Refactor OnBoarding module.

### PR DESCRIPTION
**Pull request by ruchawajpe**

### CODE REVIEW

File: lib/blocs/dateFormat/save_date_format_bloc.dart [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:toolkit/blocs/dateFormat/save_date_format_event.dart';
+import 'package:toolkit/blocs/dateFormat/save_date_format_state.dart';
+import 'package:toolkit/data/cache/customer_cache.dart';
+
+import '../../data/cache/cache_keys.dart';
+import '../../di/app_module.dart';
+
+class SaveDateFormatBloc extends Bloc<SaveDateFormatEvent, DateFormatSaved> {
```
Comment (objectorientguy) [inline lib/blocs/dateFormat/save_date_format_bloc.dart:11]: Change SaveDateFormatBloc to DateFormatBloc

File: lib/blocs/dateFormat/save_date_format_bloc.dart [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  final CustomerCache _customerCache = getIt<CustomerCache>();
+
+  SaveDateFormatBloc() : super(DateFormatSaved()) {
+    on<SaveDateFormatEvent>(_saveDateFormatValue);
+  }
+
+  FutureOr<void> _saveDateFormatValue(
+      SaveDateFormatEvent event, Emitter<DateFormatSaved> emit) async {
+    try {
+      _customerCache.setCustomerDateFormatString(
```
Comment (objectorientguy) [inline lib/blocs/dateFormat/save_date_format_bloc.dart:21]: why to we need string in the name?

File: lib/blocs/dateFormat/save_date_format_event.dart [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class SaveDateFormatEvent {
```
Comment (objectorientguy) [inline lib/blocs/dateFormat/save_date_format_event.dart:1]: Rename to DateFormatEvent

File: lib/blocs/dateFormat/save_date_format_state.dart [hunk 4]
```diff
@@ -0,0 +1 @@
+class DateFormatSaved {}
```
Comment (objectorientguy) [inline lib/blocs/dateFormat/save_date_format_state.dart:1]: Rename to DateFormatState

File: lib/screens/onboarding/selectDateFormat/select_date_format_screen.dart
Comment (objectorientguy) [inline lib/screens/onboarding/selectDateFormat/select_date_format_screen.dart:1]: Did we not try with bloc? Lets not use stateful widgets

File: lib/screens/onboarding/widgets/type_expansion_tile.dart
Comment (objectorientguy) [inline lib/screens/onboarding/widgets/type_expansion_tile.dart:1]: Try with Bloc

**Review by objectorientguy — CHANGES_REQUESTED** [review]
Requested changes

**Review by objectorientguy — CHANGES_REQUESTED** [review]

**Review by objectorientguy — APPROVED** [review]

File: lib/widgets/generic_app_bar.dart [hunk 5]
```diff
@@ -3,16 +3,17 @@ import 'package:toolkit/configs/app_theme.dart';
 import '../configs/app_dimensions.dart';
 
 class GenericAppBar extends StatelessWidget implements PreferredSizeWidget {
-  const GenericAppBar({Key? key, this.title, this.leading, this.actions})
+  const GenericAppBar(
+      {Key? key, this.textValue = '', this.leading, this.actions})
       : super(key: key);
-  final Widget? title;
+  final String textValue;
```
[the remark this answers is not in the document]
Comment (objectorientguy) [reply lib/widgets/generic_app_bar.dart:9]: still not done?

File: lib/blocs/dateFormat/date_format_states.dart [hunk 6]
```diff
@@ -0,0 +1,11 @@
+abstract class DateFormatStates {}
+
+class DateFormatInitial extends DateFormatStates {}
```
[the remark this answers is not in the document]
Comment (objectorientguy) [reply lib/blocs/dateFormat/date_format_states.dart]: hello? can you hear me?

File: lib/blocs/dateFormat/date_format_events.dart [hunk 7]
```diff
@@ -0,0 +1,7 @@
+class DateFormatEvent {
```
[the remark this answers is not in the document]
Comment (objectorientguy) [reply lib/blocs/dateFormat/date_format_events.dart]: ????

File: lib/blocs/dateFormat/date_format_bloc.dart [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+import '../../di/app_module.dart';
+
+class DateFormatBloc extends Bloc<DateFormatEvent, DateFormatStates> {
+  final CustomerCache _customerCache = getIt<CustomerCache>();
+
+  DateFormatBloc() : super(DateFormatInitial()) {
+    on<DateFormatEvent>(_saveDateFormatValue);
+  }
+
+  FutureOr<void> _saveDateFormatValue(
```
[the remark this answers is not in the document]
Comment (objectorientguy) [reply lib/blocs/dateFormat/date_format_bloc.dart]: hello??

[the remark this answers is not in the document]
Comment (ruchawajpe) [reply lib/blocs/dateFormat/date_format_bloc.dart]: This change is updated.

File: lib/blocs/dateFormat/date_format_events.dart (hunk 7 above)
[the remark this answers is not in the document]
Comment (ruchawajpe) [reply lib/blocs/dateFormat/date_format_events.dart]: This change is updated.

File: lib/blocs/dateFormat/date_format_states.dart (hunk 6 above)
[the remark this answers is not in the document]
Comment (ruchawajpe) [reply lib/blocs/dateFormat/date_format_states.dart]: This change is updated.

File: lib/widgets/generic_app_bar.dart (hunk 5 above)
[the remark this answers is not in the document]
Comment (ruchawajpe) [reply lib/widgets/generic_app_bar.dart:9]: This change is reflected.

File: lib/blocs/dateFormat/date_format_states.dart (hunk 6 above)
[the remark this answers is not in the document]
Comment (ruchawajpe) [reply lib/blocs/dateFormat/date_format_states.dart]: This change is reflected.

File: lib/blocs/dateFormat/save_date_format_bloc.dart (hunk 1 above)
Comment (ruchawajpe) [reply lib/blocs/dateFormat/save_date_format_bloc.dart]: This change is updated.

**Pull request merged**